[DOCS-7856] updating network topology map (#23056)
DataDog/documentation: content/en/network_monitoring/devices/network_topology_map.md

{{< /site-region >}}

## Overview

The [Network Device Topology Map][2] provides an overview of your network's physical connections, so you can more easily identify issues in your devices and understand their upstream and downstream impacts.

{{< img src="/network_device_monitoring/network_topology_map/network_topology_map_search_2.mp4" alt="The network device topology map, with vendor:cisco added to the search bar, and then the Filter nodes box filtered by nyc. A node is selected and the inspect option chosen, showing its connected nodes. One of the connected nodes is then selected, and the inspect option is selected again, displaying additional connected nodes" video="true" >}}

## Setup

The Datadog Agent version 7.52 and later automatically collects topology data. No additional installation is necessary.

### Prerequisites

1. Devices have LLDP (Link Layer Discovery Protocol) and/or CDP (Cisco Discovery Protocol) enabled with SNMP.
2. Datadog Agent version 7.52 or later is installed.

## Navigation options

In the Network Topology Map, the following navigation options are available:

### View

> 1. Under **View By**, select to visualize devices in groups according to tags: 

Reviewer: ```suggestion
1. In **View By**, use tags to select how you want to visualize devices: 
```

Reviewer: just a suggestion.

{{< img src="/network_device_monitoring/network_topology_map/network_view_by_location_2.png" alt="The navigation option, with view by devices and tags selected, highlighting view by location" style="width:80%;" >}}

### Color

2. Under **Color By**, change how nodes on the Device Topology Map are colored based on:

- **Device State**: Display nodes on the Device Topology Map by SNMP reachability.
- **Ping State**: Display nodes on the Device Topology Map by Ping status. 

{{< img src="/network_device_monitoring/network_topology_map/device_topology_color_by_device.png" alt="The navigation option, with view by color selected, highlighting view by device state" style="width:80%;" >}}

   The following are the definitions of the nodes for each color state:
   <div style="width:80%; margin: 0 auto;">

   | Color    | Description               |
   |----------|---------------------------|
   | Green   | Device is reachable.      |
   | Red   | Issue with device, such as unreachable through SNMP.  |

>    | Gray    | The device is monitored by NDM, however, no data has been received. For example, if the ping wasn't configured and you opted to `color by` **Ping State** in the Device Topology Map, the devices are displayed in gray. |

Reviewer: ```suggestion
   | Gray    | Device is monitored by NDM; however, no data has been received. For example, if the ping wasn't configured and you opted to `color by` **Ping State** in the Device Topology Map, the devices are displayed in gray. |
```

Reviewer: you weren't using articles at the beginning of the other Descriptions, so I removed this one.

I both Ping and ping in the doc. You might want to just use ping, whether it's a noun or a verb. Very common usage.

   | No color | Shadow devices that are not directly monitored by NDM, but are discoverable through LLDP/CDP from a connected device that NDM is monitoring. You can toggle on/off the [Hide _N_ unmonitored device section](#filter-devices) if you want these devices to be shown on the Device Topology Map.         |

   </div>

### Filter devices

3. Under **Filter Devices**, gain further granular control over what devices are shown on the Device Topology Map. 

{{< img src="/network_device_monitoring/network_topology_map/device_topology_filter_devices_hide.png" alt="The navigation option, with view filer option selected, toggling by hide unmonitored devices " style="width:80%;" >}}

> **Note:** The filter devices setting impacts what devices are shown on the Device Topology Map for _all_ queries you may make, such as if you filter by a device facet in the search bar. 

Reviewer: ```suggestion
**Note:** The **Filter Devices** setting impacts what devices are shown on the Device Topology Map for _all_ queries you might make. For example, if you filter by a device facet in the search bar. 
```

Hide _N_ Unmonitored Devices - Turned OFF by default.
: Toggling this on hides devices on the Device Topology Map that are not directly monitored by Network Device Monitoring, but still discovered by LLDP/CDP, and shown on the map from adjacent devices that are monitored by Network Device Monitoring.

Hide _N_ Unconnected Devices - Turned ON by default.
: Toggling this on hides any devices that have no link connections. Devices can be unconnected for reasons such as improper configuration, or the device does not support [LLDP/CDP](#troubleshooting).

> ### Icon legend 

Reviewer: this section is under **Navigation options**. Is it an option they can control? If not, consider moving it.

Author: Yes it's a toggle, thanks for the tip though!

{{< img src="/network_device_monitoring/network_topology_map/no_connections_found.png" alt="The no devices found message that is displayed when NDM is either not configured or because of filtering." style="width:80%;" >}}

- Turn the "Hide _N_ Unconnected Devices" toggle off to show the isolated devices.
- Use categorization tag to help understand your map view with information hierarchy.

### Empty map/ no monitored devices

- Ensure the "Hide _N_ Unconnected Devices toggle is off.

### Missing devices/connections

The Device Topology Map data is based on LLDP (Link Layer Discovery Protocol) and CDP (Cisco Discovery Protocol) information collected with SNMP. If your map is missing devices and/or connections, ensure the following:

- Datadog Agent version 7.52 or later is installed.
- Devices have LLDP and/or CDP enabled with SNMP.

Verify that your devices are exposing LLDP and CDP data with the following commands:

For LLDP data:

```yaml
sudo -u dd-agent datadog-agent snmp walk <DEVICE_IP> 1.0.8802
```
For CDP data
```yaml:
sudo -u dd-agent datadog-agent snmp walk <DEVICE_IP> 1.3.6.1.4.1.9.9.23
```

### Missing connections or links 

> If your device is exposing topology data with LLDP or CDP but some of the connections are missing, ensure that the "Hide _N_ Unmonitored Devices" toggle is off.

Reviewer: do you want to use quotes around UI elements? Typically, we bold them.

Author: let me ask the PM about this, thanks!

Use the "Hide _N_ Unmonitored Devices" toggle to hide the unmonitored nodes.

### Borderless or black nodes on the map 

The borderless or black nodes on the Device Topology Map can represent devices discovered with LLDP or CDP that are not configured to be monitored with NDM, or devices discovered with LLDP or CDP that can not be resolved to the equivalent [monitored device](#device-resolution).

## Device resolution 

The Device Topology Map provides an overview of the devices monitored with NDM and their physical connections. The topology links data is based on LLDP (Link Layer Discovery Protocol) or CDP (Cisco Discovery Protocol) information collected with SNMP. 
The connections discovered with LLDP or CDP can correspond to devices already monitored with SNMP. The device resolution consists in matching the discovered device to the monitored device.

### Device resolution failures

> The device resolution can fail if the device is not monitored with NDM, or the LLDP or CDP data is insufficient to match the discovered device to the monitored device.

Reviewer: It seems like these requirements are mentioned multiple times in the topic. Consider just mentioning them in the Prereq's and linking to that. Just easier than repeating it, IMO.
